### DISCUSSION TITLE
route ID changed

### DISCUSSION THREAD

**Discussion by akomm** [kiliman/remix-flat-routes#126] [General]
Before I open it in issue, maybe doc confusion again :)

It appears that the route IDs have been altered some. I find frequently a /route suffix at the end of the ID. 
Its not a frequent thing to use the ID of the routes, but there is one official in remix that is documented,  that being useRouteLoaderData(). But there are other use-cases to, I have one and had to adjust code to remove the suffix.

People who use this would have code breakage after adjusting the flat-routes to nested flat-routes because IDs of their routes changed.
I have adjusted my code to remove the suffix, but wanted to point out that you might have overlooked it.

**Comment by akomm** [comment]
It basically includes the route file name. _route, route, _layout, etc. Which is not the behavior of the original system.

**Reply by akomm** [reply]
I'd also like to add, that the fact that remix provides public API that requires route ID usage, it means that the route ID should be less likely a subject to arbitrary changes in remix itself. So something that would only be changed in major updates with potentially a migration notice. Otherwise I would not rely on route IDs if they could so simply and arbitrary change. Just wanted to share my mind-set on this one.

**Reply by kiliman** [reply]
Thanks for bringing up the issue. Generally, in Remix, the route id is the filename without the extension. I'll have to check against the official release to see if they strip the `route` filename for colocated routes.

To clarify, `remix-flat-routes` came **before** the v2 routing convention, so they may have done something different.

**Reply by akomm** [reply]
I understand. I can definitely say that before there was no /route at the end, because a very crucial part of my app relied on it. But its not a big deal adjusting. My logic skips segments in the id with _ anyway, so that covers all the _about _route, etc. and I have added a skip for the last segment if its one of the index, route, layout, page - obviously it then can't be part of route, but not intended by me.

If I understood correct, the flat one with +nested was mentioned as BC to original with extra, that is why I thought maybe it was overlooked. But there is no drama on my side :)

**Reply by kiliman** [reply]
Ok, I think I see the issue. Yes, it appears that `v2` routing doesn't include the `route` as part of the id.

Dumps from `npx remix routes --json` (json format includes id)

## v2 routing
```json
[
  {
    "id": "root",
    "path": "",
    "file": "root.tsx",
    "children": [
      {
        "id": "routes/colocated",
        "path": "colocated",
        "file": "routes/colocated/route.tsx"
      },
      {
        "id": "routes/_index",
        "index": true,
        "file": "routes/_index.tsx"
      }
    ]
  }
]
```

## `remix-flat-routes`
```json
[
  {
    "id": "root",
    "path": "",
    "file": "root.tsx",
    "children": [
      {
        "id": "routes/_index",
        "index": true,
        "file": "routes/_index.tsx"
      },
      {
        "id": "routes/colocated/route",
        "path": "colocated",
        "file": "routes/colocated/route.tsx"
      }
    ]
  }
]
```

That is a bug. I'll fix it, but I'll probably add a config option to enable the fix as I don't want to change route ids for those people that depend on the current behavior.

**Reply by kiliman** [reply]
Added issue #128
